### PR TITLE
Response streaming and dependency updates

### PR DESCRIPTION
**Pull request by jskeet**
Once this is in, I'll need to update https://github.com/googleapis/google-cloud-dotnet/pull/3719 appropriately.

### CODE REVIEW

File: Google.Api.Gax.Grpc/BidirectionalStreamingBase.cs [hunk 1]
```diff
@@ -110,11 +111,16 @@ public virtual Task WriteCompleteAsync()
         // Streaming responses
 
         /// <summary>
-        /// Async stream to read streaming responses.
+        /// Async stream to read streaming responses, exposed as an async sequence.
+        /// The default implementation will construct use <see cref="GrpcCall"/> to extract a response
```
Comment (chrisdunelm) [inline Google.Api.Gax.Grpc/BidirectionalStreamingBase.cs]: Typo "construct use"

File: Google.Api.Gax.Grpc/ResponseStreamAdapter.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        internal ResponseStreamAdapter(IAsyncStreamReader<TResponse> reader, CancellationToken cancellationToken)
+        {
+            _reader = reader;
+            _cancellationToken = cancellationToken;
+        }
+
+        public TResponse Current => _reader.Current;
+
+        // The gRPC response stream doesn't support disposal. The call does, but users should dispose that themselves
```
Comment (chrisdunelm) [inline Google.Api.Gax.Grpc/ResponseStreamAdapter.cs]: Nit: It's not immediately clear what is meant by "The call does"

File: Google.Api.Gax.Grpc/ServerStreamingBase.cs [hunk 3]
```diff
@@ -26,11 +28,16 @@ public virtual AsyncServerStreamingCall<TResponse> GrpcCall
         }
 
         /// <summary>
-        /// Async stream to read streaming responses.
+        /// Async stream to read streaming responses, exposed as an async sequence.
+        /// The default implementation will construct use <see cref="GrpcCall"/> to extract a response
```
Comment (chrisdunelm) [inline Google.Api.Gax.Grpc/ServerStreamingBase.cs]: Typo; same as above "construct use".

File: Google.Api.Gax.Grpc/BidirectionalStreamingBase.cs (hunk 1 above)
Comment (jskeet) [reply Google.Api.Gax.Grpc/BidirectionalStreamingBase.cs]: Thanks, will fix.

File: Google.Api.Gax.Grpc/ResponseStreamAdapter.cs (hunk 2 above)
Comment (jskeet) [reply Google.Api.Gax.Grpc/ResponseStreamAdapter.cs]: Righto, will amend.

**Pull request merged**